### PR TITLE
ActiveDataProvider should avoid query data when data count is zero

### PR DESCRIPTION
**Pull request by kLkA**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #12752

### CODE REVIEW

File: framework/data/ActiveDataProvider.php [hunk 1]
```diff
@@ -103,7 +103,9 @@ protected function prepareModels()
         }
         $query = clone $this->query;
         if (($pagination = $this->getPagination()) !== false) {
-            $pagination->totalCount = $this->getTotalCount();
+            if (($pagination->totalCount = $this->getTotalCount()) === 0) {
```
Comment (Renkas) [inline framework/data/ActiveDataProvider.php]: why put all the logic in one line? Good practice is that every action goes to a separate line. Here you are doing assignment and then comparison on one ...

I can't see any performance optimization neither on this syntax.

**Review by samdark — APPROVED** [review]

File: framework/data/ActiveDataProvider.php (hunk 1 above)
Comment (kLkA) [inline framework/data/ActiveDataProvider.php]: Hi @Renkas , I agree with you. Thank you

**Review by SilverFire — CHANGES_REQUESTED** [review]
The test does not check the query skipping fact. It just proves, that you will get 0 modes when count query returns 0 rows.

**Comment by SilverFire** [comment]
My suggestions is to create a `ZeroCountQueryStub` class, with `count()` method that always returns 0

**Comment by dynasource** [comment]
> The test does not check the query skipping fact. It just proves, that you will get 0 modes when count query returns 0 rows.

agree

**Pull request closed without merging**